### PR TITLE
fix: handle missing contract type JSONs [APE-650]

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Follow up to https://github.com/ApeWorX/ape/pull/1298

Right now, if you migrate from 0.6.2 to the soon-to-be 0.6.3, you will get issues where Ape cannot find your contracts because of some recent in-memory caching advancements. This PR does 2 things:

1. Allows migration to 0.6.3 by searching for the missing contract and recompiling it
2. Handles the case where if you manually delete a file JSON from your .build cache, it will re-compile it and give it back to you.

### How to verify it

* Install ape 0.6.2 and compile a project
* Switch to ape env with this branch installed
* Try to run a script or tests that uses one of the compiled scripts
* Before, it would fail saying the contract was not found, which may be confusing. Now, it will seemlessly work like nothing every happened (migration).
Additionally, if you delete a contract type JSON, Ape will now notice that and recompile them and things will still work as well.

**Basically, there is no reason to get a `ContractTypeNotFound` error in a local project for your own contracts when they actually exist.**

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by unparalleled-js** [comment]
trying to find a better approach

File: src/ape/api/projects.py [hunk 1]
```diff
@@ -95,11 +95,29 @@ def cached_manifest(self) -> Optional[PackageManifest]:
...
-            self._cached_manifest = manifest
-        return self._cached_manifest
+            self._cached_manifest = _load_manifest_from_file(self.manifest_cachefile)
+            if self._cached_manifest is None:
+                return None
+
+        manifest = self._cached_manifest
+        if manifest.contract_types and not self.contracts:
+            # Extract contract types from cached manifest.
+            # This helps migrate to >= 0.6.3.
```
Comment (fubuloubu) [inline src/ape/api/projects.py:105]: Attitude note to remove in 0.7.0

**Review by fubuloubu — DISMISSED** [review]

File: src/ape/api/projects.py (hunk 1 above)
Comment (unparalleled-js) [reply src/ape/api/projects.py:105]: OMG, "Add a TODO note" = "Attitude note", always now.

Comment (fubuloubu) [reply src/ape/api/projects.py:105]: lmfao whoops

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]
